src/utils.ts

```diff
@@ -182,25 +188,62 @@ export const bufferToBase64UrlEncoded = input => {
   );
 };
 
-const getJSON = async (url, options) => {
-  const response = await fetch(url, options);
+const fetchWithTimeout = (url, options, timeout = DEFAULT_FETCH_TIMEOUT_MS) => {
+  // The promise will resolve with one of these two promises (the fetch and the timeout), whichever completes first.
+  return Promise.race([
+    fetch(url, options),
+    new Promise((_, reject) => {
+      setTimeout(
+        () => reject(new Error("Timeout when executing 'fetch'")),
+        timeout
+      );
+    })
+  ]);
+};
+
+const getJSON = async (url, timeout, options) => {
+  let fetchError, response;
+
+  for (let i = 0; i < DEFAULT_SILENT_TOKEN_RETRY_COUNT; i++) {
```

Comment:
seems like `DEFAULT_SILENT_TOKEN_RETRY_COUNT` is hardcoded and not customizable by the user. Do we want it that way or should we open an option value here?

Reply:
It feels like something we can leave at a sensible default and see what feedback we get around it. I can't see people needing to change this value too much.